src/components/SideBar/SideBar.tsx

```diff
@@ -1,29 +1,45 @@
-import React, { useRef } from 'react'
 import {
-  ChevronDoubleLeftIcon,
-  ExclamationIcon,
-  ClockIcon,
-  ExclamationCircleIcon
-} from '@heroicons/react/outline'
+  MutableRefObject,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState
+} from 'react'
+import { ExclamationIcon, ChevronLeftIcon } from '@heroicons/react/outline'
+import { composeClasses } from 'lib/classes'
 import Text from '../Typography'
 import ToolTipHover from '../ToolTipHover'
-import './sideBar.css'
-
-import { composeClasses } from 'lib/classes'
 import Flex from '../Layout/Flex/Flex'
-import Skeleton from '../Skeleton/Skeleton'
+import SideBarHeader from './SideBarHeader'
+import SideBarItem from './SideBarItem'
+import SkeletonSideBarList from './SkeletonSideBarList'
+import './sideBar.css'
 
-export interface SideBarProps {
-  sideBarList?: Array<{
+export interface ISideBarSubItem {
+  [key: string]: {
     title: string
     active: boolean
     to: () => void
-    icon?: JSX.Element
-    disabled?: boolean
-    hidden?: boolean
-  }>
+  }
+}
+
+export type TBadge = string | number | ReactElement
+export interface ISideBarItem {
+  title: string
+  active: boolean
+  isOpen?: boolean
+  to?: () => void
```

Comment:
I think we can change the name to other more explicative

```suggestion
  onClick?: () => void
```

Reply:
I think it may be a bit confusing as this function will not be called every time the element is clicked, it will only be called if the parent element has no sub-elements